app/src/main/java/app/grapheneos/camera/capturer/ImageSaver.kt

```diff
@@ -73,17 +74,33 @@ class ImageSaver(
     val contentResolver = appContext.contentResolver
     val mainThreadExecutor = appContext.mainExecutor
 
+    private var isCancelled = false
+
+    fun cancelCaptureRequest() {
+        isCancelled = true
+    }
+
     override fun onCaptureSuccess(image: ImageProxy) {
-        mainThreadExecutor.execute(imageCapturer::onCaptureSuccess)
+        mainThreadExecutor.execute mainExecutor@ {
+            if (isCancelled) return@mainExecutor
```

Comment:
What's the point of this change? Why prevent already captured image from being saved?

Reply:
This change was added as a safety check to deal with the app going in background (via onPause) as the remaining execution could get uncertain since the app is already in background if the cancelled flag is set to true

Reply:
What do you mean by "execution could get uncertain"?

Reply:
I mean the certainty of the proper completion of the execution

Reply:
There's never such certaintly. App process can be stopped at any time, including when the app is the top foreground app.

Reply:
Consider what happens after this change when the Camera app activity gets paused due to another activity being launched on top by the OS. That happens automatically in some cases. If such launch happens right after image capture then the image would get discarded.

Reply:
That's true actually. I'll revert the changes to continue saving the image once a capture is successful as before

Reply:
Have pushed the changes. Thanks for reviewing!